tests/utils/test_tokenizers.py

```diff
@@ -63,3 +66,40 @@ def test_out_of_range_ascii():
     # characters out of ascii range are mapped to zero (null)
     assert tokens[0] == 0
     assert tokens[1] == 0
+
+def test_single_token():
+    """
+    checks if convert_tokens_to_ids handles both single strings and lists. single tokens
+    should all be converted into a list of one id.
+    """
+    # testing character tokenizer
+    char_tokenizer = get_tokenizer('char_tokenizer')
+    assert char_tokenizer.convert_tokens_to_ids("h") == [104]
+    assert char_tokenizer.convert_tokens_to_ids("e") == [101]
+    assert char_tokenizer.convert_tokens_to_ids(["h", "e", "l", "l", "o"]) == [
+        104,
+        101,
+        108,
+        108,
+        111,
+    ]
+    # testing sentencePiece tokenizer
+    with tempfile.TemporaryDirectory() as workdir:
+        training_data = os.path.join(workdir, 'training_data.txt')
+        with open(training_data, 'w', encoding='utf-8') as f:
+            f.write('This is a test.\n')
+            f.write('Please say hello world.')
+        model = os.path.join(workdir, 'mymodel')
+        spm.SentencePieceTrainer.train(f'--input={training_data} --model_prefix={model} --vocab_size=76, --model_type=bpe --character_coverage=1.0')
+        sp = spm.SentencePieceProcessor()
+        sp.load(f'{model}.model')
+        sp_tokenizer = get_tokenizer(name=f'{model}.model', style='sentencepiece')
+        assert sp_tokenizer.convert_tokens_to_ids("h") == [66]
```

Comment:
since a string is effectively an array of characters, I suspect this test might pass even without the change. We probably need a test that captures at least some multi-character strings. One quick way to do that might be to add some multi-character 'special' tokens to the tokenizer.

Reply:
Without the change, the assert statement on line 97 errored because it returned 66 = [66], but I will test some multi-character strings to make sure that those work too.